gluten-core/src/main/java/org/apache/gluten/vectorized/JniWorkspace.java

```diff
@@ -75,14 +75,14 @@ private static JniWorkspace createDefault() {
     }
   }
 
-  public static void enableDebug() {
+  public static void enableDebug(String debugDir) {
```

Comment:
What happens if `debugDir` changed by user in session?

We either handle the case or just raise an error (by making the option static or adding assertion in code).

Reply:
@Yohahaha Let's still add an assertion to make sure the function doesn't get call twice with different `debugDir` values. In case the method is used in test code or somewhere. Thanks.

That may involve adding another global variable but it could be worth it I guess.

Reply:
so you think `DEBUG_INSTANCE` will be initialized twice in one process? if yes, we should refactor jni initialization first.

Reply:
> so you think DEBUG_INSTANCE will be initialized twice in one process

Noop. Just a protection in this method assuming unpredictable caller code. E.g., One could call `enableDebug(...)` for test purpose. We'd throw something otherwise the second call will be silently swalloed.

Reply:
this functionality should only be called once by design from my understanding, if someone want call it twice, I think they dont know that they did. We dont really need protect anything.
Related conf already be static and `enableDebug` has ensure only initialize `DEBUG_INSTANCE` once.

Reply:
> I think they dont know that they did. We dont really need protect anything.

I think anti cases are appearing easier than we think it could be.

For example, if someone in test code, calling `JniWorkspace.enableDebug("folder1") for debug purpose`, however in somewhere a Spark config "spark.gluten.sql.debug.keepJniWorkspaceDir="folder2" was already set for some reason.

Then he could go mad because the files was not found in folder1 and the console doesn't tell him anything about that.

Reply:
As a debug conf, I feel that we don't need to provide excessive protection for the user.

Reply:
`enableDebug("folder1")` should just enable debug in folder `folder1`. If there's potential that it doesn't use `folder1`, please give user/caller a note.

I don't think it's excessive protection. Next time please close comments like this one before merging a PR.